### ISSUE TITLE
How can I add TAGs to the Add a fact lists - wt 2.1.0-dev

### ISSUE THREAD

**Issue by melizaa** [fisharebest/webtrees#4090]
I see certain TAGs for instance on individual pages. How can I add them for instance to new or other existing Individuals?

I do not want to include all the TAGs defined in the CustomTags, etc. I want to select the relevant ones for my site.

**Comment by melizaa** [comment]
Greg,

The old My local languages module does not work in 2.1.0.
I saw it in the Control panel / All modules and unchecked the module.

I added the old texts and some new customTags and customSubTags to ExampleModuleCustomTags.php.
I commented out the handling of the module version.
I transferred the webtrees_example_module_custom_tags directory to the future test system's modules_v4. 
I hope that I did not make any mistakes.

I do not see "Custom tags" in Control panel / All modules.

**Comment by fisharebest** [comment]
It would be so much easier to help if I could view/edit the code directly.

Please let me create a shared github repository for your module(s).

I could probably have fixed the problems in less time that it took me to write this reply.

**Comment by melizaa** [comment]
OK, thanks.

**Comment by melizaa** [comment]
I have at least 2 local code changes that existed in 1.8, but have vanished in 2.0.
I do not understand the 2.0 code well enough to change the code locally.
1. I added to the top and bottom of the list pages the code of "Showing nn to nn of nn" and the pagination.
This helps me when I look at long lists, I do not have to go to the top of each page in order to go to the next page.
2. I added to the individual lists for administrators next to the sex icon ** for site users and @ for other site individuals who have an email in their data. This helps me contact family members of a certain individual.

I hope that you can point me to the correct code.

Should I create new issues for these?

**Comment by melizaa** [comment]
Regarding the two subject modules, I do not know where I should add them on the server?
I think that I will manage to get them working.

**Comment by melizaa** [comment]
Greg,

Thanks!

I had to cut and paste your file texts to my local site files before I transferred the test site files to the future site's modules_v4.
When I tried to define to use TortoiseGit I could not log in to Github because of new security rules.

I see now the custom tags, the translations, and the pagination controls.

I do not see the
  adds ** for individuals linked to an administrator
  adds @ for other individuals with an email in their data

I do not see all the facts in the Add facts select lists.
I assume that I should add them to the select lists in the Gedcom Preferences by filling them in the Facts for individual records and Facts for family records.

[Have you seen this site?](https://wiki-de.genealogy.net/GEDCOM/_Nutzerdef-Tag)
Some of the Tags that we added as custom Tags to my site exist there.

**Comment by melizaa** [comment]
Greg, good morning,

I did not finish my testing.
I think that I saw some missing texts.

Today I will attend a tour,
Meliza

**Comment by melizaa** [comment]
Hello Greg,

I update regularly the two test systems with the wt code changes and I try
to update the vendor files.



There are some issues. How can I solve them?



1. How can I add aka names or married names to 1 NAMEs and remove the name
type? INDI:NAME:_AKA & INDI:NAME:_MARNM



2. I do not want as a default INDI:BIRT Y and INDI:DEAT Y.
    Users should be able to fill this value for death if they know that a
person is dead, but not when he died. All the site individuals were born,
so we do not need INDI:BIRT Y.



3. I do see FAM:_ADPF and FAM:_ADPM, but not the translated texts for
instance in the big Gedcom for F247/F95 and F13410.
    I cannot add these facts to an individual or a family.



4. How would you update the Not Married 1 INDI:_NMR 2 DATE in the Gedcom?

    I have updated the Never married INDI:_NMAR to Number of marriages 0.



5. INDI:DEAT:AGE CHILD is not translated to the site languages. AGEs YOUTH,
MAIDEN, WOMAN and YOUNG are translated – see for instance I21873 and I24282.



6. I try to add an associate to a new Research task. I see
*INDI:_TODO:ASSO:NOTE
*and* INDI:_TODO:ASSO:SOUR *instead of Note (or TEXT) and Source texts. The
note and source data are not added to the Gedcom file.



7. I cannot add the additional subordinate address structure data
INDI:ADDR:ADR1, INDI:ADDR:ADR2, INDI:ADDR:ADR3, INDI:ADDR:CITY,
INDI:ADDR:CTRY, INDI:ADDR:POST, INDI:ADDR:STAE
I can neither add INDI:FAX
The GEDCOM 5.5.1 defines these facts.



8. I want also to add INDI:DATA:DATE, INDI:DATA:SOUR, INDI:DATA:SOUR:TEXT &
FAM:DATA:DATE, FAM:DATA:SOUR, FAM:DATA:SOUR:TEXT



9. I want to add SOUR:AUTH and SOUR:AUTH:NOTE and SOUR:NOTE.
    I have filled in production in the NOTEs for instance [Roni
Klimscheffskij - רוני קלימצווסקי](
http://www.amitys.com/webtrees/individual.php?pid=I191&ged=Gedcom.ged)
I need a utility to update this probably to [Roni Klimscheffskij - רוני
קלימצווסקי](
https://www.amitys.com/webtrees/index.php?route=%2Ftree%2FGedcom.ged%2Findividual%2FI191%2FRoni-Klimscheffskij
)



10. In the monthly calendar I see in production Sunday as the leftmost
week-day. In the test systems I see Monday as the leftmost day. How can I
correct this?

11. When I search for R13 (Family Trees/Books – Restricted) on the top
search I get error
Undefined variable: linked_submitters …/resources/views/record-page.phtml:54
#0 …/resources/views/record-page.phtml(54):
Fisharebest\Webtrees\Webtrees::Fisharebest\Webtrees\{closure}()
etc.

12. How/when can I see a list of the missing texts of facts and other
GEDCOM errors?
Now I have detected some in a fluke.

13. I will probably detect more issues when I continue to test the code.



14. There are many changes that I will have to carry out manually in my
Gedcom file to get rid of data that is no longer supported by the new
webtrees version.



15. vendor files



cd /mnt/c/webtreesGIT

php /home/meliza/composer.phar install –no-dev

gives the following comment since the new tecnickcom/tcpdf (6.4.4) version.
Invalid argument –no-dev. Use "composer require –no-dev" instead to add
packages to your composer.json.



So, I use php /home/meliza/composer.phar update.

This updates also the development packages.

When I look at the composer.lock file I can more or less figure out what I
should have on the server test sites.

Do I need the following in the server vendor directories?

·       bin files

·       composer files, not directories



I assume that I do not need the following on the server?

·       composer directories?

·       doctrine

·       justinrainbow

·       league

·       myclabs

·       nikic

·       phar-io

·       php-coveralls

·       phpdocumentatior

·       phpspec

·       phpunit

·       react

·       sebastian

·       seld

·       symfony

·       system

·       theseer

·       webmozart



I see also

As there is no 'unzip' command installed zip files are being unpacked using
the PHP zip extension.


Meliza

**Comment by fisharebest** [comment]
> How can I add aka names or married names to 1 NAMEs and remove the name
type? INDI:NAME:_AKA & INDI:NAME:_MARNM

The GEDCOM standard is `1 NAME ... / 2 TYPE aka`.
The custom tag `1 NAME ... / 2 _AKA ...` was created by some old desktop programs.

PGV used the custom tags, and so webtrees inherited that.
But we are moving to the GEDCOM standard tags.
Both are supported, but you are encouraged to use the standard format now.

There is a datafix to convert to the GEDCOM standard.

Are you saying that you do not want to use the GEDCOM standard, and that you want to use the custom `_AKA` tags?

> I do not want as a default INDI:BIRT Y and INDI:DEAT Y.

That may be a bug.  When we add a new fact, we insert the `Y`.

For adding new facts for new individuals, this is probably not required.

> I do see FAM:_ADPF and FAM:_ADPM, but not the translated texts for instance in the big Gedcom for F247/F95 and F13410. I cannot add these facts to an individual or a family.

Adding custom tags hasn't been written yet.  Nothing difficult.  All the code is there.  Just need to work out which tags go where.

> How would you update the Not Married 1 INDI:_NMR 2 DATE in the Gedcom?

As above, knowing which custom tags goes where still needs to be done.

> INDI:DEAT:AGE CHILD is not translated to the site languages. AGEs YOUTH, MAIDEN, WOMAN and YOUNG are translated – see for instance I21873 and I24282.

`CHILD` wasn't listed as an option in your module.  I have just added it:

https://github.com/fisharebest/amitys-dot-com/commit/ca5ba70d7f0fdca6a17bc580a4795ede8fc3e165

> I cannot add the additional subordinate address structure data

Some tags are infrequently used or not recommended.  These are hidden initially.
In the latest code, you see them by clicking the button "Edit with all GEDCOM tags".

> I want also to add INDI:DATA:DATE, INDI:DATA:SOUR, INDI:DATA:SOUR:TEXT &
FAM:DATA:DATE, FAM:DATA:SOUR, FAM:DATA:SOUR:TEXT

These custom tags are defined in your module.  I just addeded them.

See https://github.com/fisharebest/amitys-dot-com/commit/bf8b94607dded6496522ab1702870a258f01ff2e

> I want to add SOUR:AUTH and SOUR:AUTH:NOTE and SOUR:NOTE.

You should already be able to add these.

SOUR:AUTH and SOUR:NOTE are standard GEDCOM.

SOUR:AUTH:NOTE is defined in your module https://github.com/fisharebest/amitys-dot-com/blob/bf8b94607dded6496522ab1702870a258f01ff2e/module.php#L277

> In the monthly calendar I see in production Sunday as the leftmost
week-day. In the test systems I see Monday as the leftmost day. How can I
correct this?

The start-of-week day is defined by your language.  For example, British English uses Monday, while American English uses Sunday.

We use a official localization database for this. (https://cldr.unicode.org/)

To override this, you'd need to modify this line:
https://github.com/fisharebest/webtrees/blob/ffee945e94939397dca2512e4d690f7d1e4707e8/app/Http/RequestHandlers/CalendarEvents.php#L169

> When I search for R13 (Family Trees/Books – Restricted) on the top
search I get error

Thanks - fixed.

> How/when can I see a list of the missing texts of facts and other GEDCOM errors?

This is planned.  Now that we have GEDCOM definitions built-in to the code, we should be able to use them to create a validation/check.  The code will be simple.  But coping with large trees and low memory/cpu limits could be challenging.

> There are many changes that I will have to carry out manually in my Gedcom file to get rid of data that is no longer supported by the new webtrees version.

If these are customisations that you want to keep, then we can add definitions for them in your module.

> gives the following comment since the new tecnickcom/tcpdf (6.4.4) version. Invalid argument –no-dev. Use "composer require –no-dev" instead to add packages to your composer.json.

A type.  You need two `-`.  e.g. `--no-dev`.

> Do I need the following in the server vendor directories?

No - but they are small and harmless.

> I assume that I do not need the following on the server?

Most of these are only used for development.

It may be simpler to delete all the `/vendor` folder, and then just upload the known files.

PS - the "upgrade wizard" will automatically delete old files from `/vendor`, etc.

**Comment by melizaa** [comment]
Thanks for your fast handling and answer.

1. I know.
    In my prod system, I have a bad experience with several names with the
TYPE because when updating, their sort sequence changed, so I prefer to
have a Latin name with sub-names, a Hebrew name with sub-names, and
sometimes an additional Thai/Russian, etc. name. I changed lately my prod
data to have this format.

n. Some issues are solved

15. So far I got rid of the development vendor files. I wonder how the --
was replaced by -. 😞

Meliza
